lowtime/solver.py

```diff
@@ -465,20 +567,58 @@ def find_min_cut(self, dag: nx.DiGraph) -> tuple[set[int], set[int]]:
             else:
                 residual_graph.add_edge(v, u, capacity=capacity)
 
+        profiling_min_cut_between_max_flows = (
+            time.time() - profiling_min_cut_between_max_flows
+        )
+        logger.info(
+            "PROFILING PhillipsDessouky::find_min_cut between max flows time: %.10fs",
+            profiling_min_cut_between_max_flows,
+        )
+
         # Run max flow on the new residual graph.
         try:
-            _, flow_dict = nx.maximum_flow(
-                residual_graph,
-                source_node,
-                sink_node,
-                capacity="capacity",
-                flow_func=edmonds_karp,
+            # profiling_max_flow = time.time()
+            # _, flow_dict = nx.maximum_flow(
+            #     residual_graph,
+            #     source_node,
+            #     sink_node,
+            #     capacity="capacity",
+            #     flow_func=edmonds_karp,
+            # )
+            # profiling_max_flow = time.time() - profiling_max_flow
+            # logger.info(
+            #     "PROFILING PhillipsDessouky::find_min_cut maximum_flow_2 time: %.10fs",
+            #     profiling_max_flow,
+            # )
+            profiling_max_flow = time.time()
+            nodes, edges = format_rust_inputs(residual_graph)
+
+            profiling_data_transfer = time.time()
+            rust_dag = lowtime_rust.PhillipsDessouky(
+                nodes, source_node, sink_node, edges
+            )
+            profiling_data_transfer = time.time() - profiling_data_transfer
+            logger.info(
+                "PROFILING PhillipsDessouky::find_min_cut data transfer 2 time: %.10fs",
+                profiling_data_transfer,
             )
+
+            rust_flow_vec = rust_dag.max_flow()
```

Comment:
Did you actually already replace the second max flow with Rust? How much speed up did you see?

Reply:
I was getting some errors when I replaced the second call... will look into it further.

Reply:
Successfully replaced second max flow with Rust. Summarizing all profiling results here: https://docs.google.com/spreadsheets/d/1CS7ba075mFsgYYo8KnIve6Tm_-cpZev2Lxp0kNV8bpU/edit?gid=0#gid=0